frontend/src/components/browser/clipBrowser.js

```diff
@@ -2,26 +2,150 @@ import React, { Component } from "react";
 import { connect } from "react-redux";
 import styles from "./browser.module.css";
 
+import {
+  changeMode,
+  changeBrowserTab,
+  INSPECTOR_MODE_CLIP,
+  INSPECTOR_MODE_CAMERA,
+} from "../../state/stateBrowser";
+
 // React Bootstrap
 import ListGroup from "react-bootstrap/ListGroup";
+import Collapse from "react-bootstrap/Collapse";
+import Button from "react-bootstrap/Button";
 
 /* -- Browser -- */
 class ClipBrowser extends Component {
   constructor(props) {
     super(props);
+
+    this.state = {
+      cameras: {},
+    };
+
+    this.playClip = this.playClip.bind(this);
+  }
+
+  /**
+   * This function is the callback for whenever the
+   * play button of a specific clip was clicked.
+   * 
+   * @param {Number} id The unique identifier of the clip to be played. Should be an integer.
+   */
+  playClip(id) {
+    console.log("Play clip with ID " + id);
+  }
+
+  /**
+   * Checks whether or not the camera with the specified ID
+   * has an expanded clip list or not.
+   * 
+   * @param {Number} id The unique identifier of the camera whose clip list should be checked.
+   * @return {boolean} true if the camera's clip list is expanded.
+   */
+  isToggled(id) {
+    return id in this.state.cameras ? this.state.cameras[id] : false;
+  }
+
+  /**
+   * Toggles the specified camera's clip list between collapsed and expanded.
+   * 
+   * @param {Number} id The unique identifier of the camera whose clip list should be toggled.
+   */
+  toggle(id) {
+    this.setState({
+      cameras: {
+        ...this.state.cameras,
+        [id]: id in this.state.cameras ? !this.state.cameras[id] : true,
+      },
+    });
   }
 
   render() {
     return (
       <div className={styles.browserClip}>
-        <ListGroup variant="flush">
+        <ListGroup>
+          {/* Iterate through cameras */}
           {Object.values(this.props.cameras).map((camera) => (
-            <ListGroup.Item key={camera.id}>
-              <p>{camera.name}</p>
-              {Object.values(camera.clips).map((clip) => (
-                <ListGroup.Item key={clip.id}>{clip.name}</ListGroup.Item>
-              ))}
-            </ListGroup.Item>
+            <div key={camera.id}>
+              {/* Camera item */}
+              <ListGroup horizontal>
+                {/* Camera header */}
+                <ListGroup.Item
+                  action
+                  variant="success"
+                  onClick={() => this.toggle(camera.id)}
+                  aria-controls={"cam" + camera.id}
+                  aria-expanded={this.isToggled(camera.id)}
+                >
+                  {camera.name}
+                  <span
+                    style={{
+                      position: "absolute",
+                      right: "1em",
+                    }}
+                  >
+                    {Object.keys(camera.clips).length} clips
+                  </span>
```

Comment:
Suggest add a comment what this do or purpose i.e. {/* Displays nr of clips in camera */}